irohad/ordering/impl/ordering_service_transport_grpc.cpp

```diff
@@ -84,4 +80,6 @@ void OrderingServiceTransportGrpc::publishProposal(
 OrderingServiceTransportGrpc::OrderingServiceTransportGrpc(
     std::shared_ptr<network::AsyncGrpcClient<google::protobuf::Empty>>
         async_call)
-    : async_call_(std::move(async_call)) {}
+    : async_call_(std::move(async_call)),
+      batch_factory_(std::make_unique<
+                     shared_model::interface::TransactionBatchFactoryImpl>()) {}
```

Comment:
Please pass it from `application.cpp` instead, so that impl is only included in `application.cpp`.